src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -25,6 +25,31 @@ public abstract class TrainCatalogBase
         [BestFriend]
         internal IHostEnvironment Environment => Host;
 
+        /// <summary>
+        /// A pair of datasets, for the train and test set.
+        /// </summary>
+        public struct TrainTestData
+        {
```

Comment:
Why struct?

Reply:
Why class?
it's a small collection of immutable objects.

---
In reply to: [255723990](https://github.com/dotnet/machinelearning/pull/2507#discussion_r255723990) [](ancestors = 255723990)